SLING-8523 : Duplicate OSGI configs after upgrade

ashokpanghal opened the pull request:
Upgrade resulting in duplicate factory configs for cases where alais is null

ashokpanghal commented:
@cziegeler , please review

cziegeler commented on src/main/java/org/apache/sling/installer/factories/configuration/impl/ConfigUpdateHandler.java:
```diff
@@ -77,27 +77,57 @@ public void update(final Collection<UpdatableResourceGroup> groups) {
 
-        final String factoryPid = alias.substring(0, pos - 1);
-        final String pid = oldId.substring(factoryPid.length() + 1);
+            String factoryIdString = oldId.substring(0,oldId.lastIndexOf('.')+1); // keep it +1 to have last dot intact so that we always have even dots in the string
```
you could store lastIndexOf in an int variable and reuse it below

cziegeler commented on src/main/java/org/apache/sling/installer/factories/configuration/impl/ConfigUpdateHandler.java:
```diff
@@ -77,27 +77,57 @@ public void update(final Collection<UpdatableResourceGroup> groups) {
-        final String pid = oldId.substring(factoryPid.length() + 1);
+            String factoryIdString = oldId.substring(0,oldId.lastIndexOf('.')+1); // keep it +1 to have last dot intact so that we always have even dots in the string
+            factoryPid = oldId.substring(0,getMiddleDotSplitIndex(factoryIdString,'.'));
+            pid = oldId.substring(oldId.lastIndexOf('.')+1,oldId.length());
```
substring(int) is enough, if you only specify the start it goes until the of the string

ashokpanghal commented:
@cziegeler , Thanks for the review, added a test case to cover the new code and incorporated your code comments. Please merge. 
On a side note, sonar code analysis is still showing a failure despite 100% code coverage for newly added code
